### PR TITLE
Changed standardize_and_scale to a more standard form

### PR DESCRIPTION
**Pull request by emackev**
Results are still between -1 and 1, but not relying on min/max calculations. Checked behavior (fipsqueries), looks quite similar. I prefer this version for explainability (zscore then pass through sigmoid to make data between -1 and 1). Checked it gives very intuitive results (e.g. returning kins are college towns when you enter hampshire ma).

### CODE REVIEW

File: cities/utils/cleaning_utils.py [hunk 1]
```diff
@@ -13,36 +13,24 @@ def standardize_and_scale(data: pd.DataFrame) -> pd.DataFrame:
...
-        if data.dtypes[column] != "float64":
-            new_data[column] = data[column].copy()
-        else:
-            new = data[column].copy().values.reshape(-1, 1)
-            new = standard_scaler.fit_transform(new)
-
-            positive_mask = new >= 0
-            negative_mask = new < 0
-
-            min_positive = np.min(new[positive_mask])
```
[the remark this answers is not in the document]
Comment (azane) [reply cities/utils/cleaning_utils.py:29]: I see. I've been missing that nuance.

File: cities/utils/cleaning_utils.py [hunk 2]
```diff
@@ -13,36 +13,24 @@ def standardize_and_scale(data: pd.DataFrame) -> pd.DataFrame:
...
+        return range_minus1_1
+
+    # Copy all columns first
+    new_data = data.copy()
+
+    # Select float columns
+    float_cols = data.select_dtypes(include=["float64"])
+
+    # Standardize float columns to mean 0, std 1
+    standardized_floats = standard_scaler.fit_transform(float_cols)
```
[the remark this answers is not in the document]
Comment (azane) [reply cities/utils/cleaning_utils.py:30]: I see. The [-1,1] domain requirement sounds like something that could change as well then. Could anyone link me — if easy — to where that requirement is being enforced and/or expected?

[the remark this answers is not in the document]
Comment (emackev) [reply cities/utils/cleaning_utils.py:30]: I'm not sure it's literally required, it's just the type of data we've been working with.  It'll be good to rethink more broadly (I agree it'd be good to move beyond the [-1, 1] expectation), but in terms of correcting the current min-max version, it makes sense to keep data quite similar to what we've been working with at first. I could imagine, for example, different features effectively having more or less impact on the results than they currently do, or euclidean distance giving different answers etc, with different scalings -- it'd probably all be fine and good, just being cautious.

[the remark this answers is not in the document]
Comment (rfl-urbaniak) [reply cities/utils/cleaning_utils.py:30]: I don't think moving beyond [-1,1] is safe. For similarity calculations, changing scale is what effectively weighing does, so if you allow both variable weighting and manual scale changing, you allow for the possibility that an unaware user changes weights of the same thing twice without understanding that that's what they do. Moreover, sameness of scale is important downstream for learning and interpretability. What are good reasons to allow the user to manually change ranges?

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
- Since the previous target has been merged into `main`, this PR now targets `main`.

- This PR has a `fips_query.py` conflict, which looks like it's related to @riadas modification to plotting functions. Ria, can you resolve it to your liking?

- Once the conflict is resolved, please `pull origin main` after making sure your local copy of main is up to date. Then run the test.

- As the PR stands, it fails tests locally. Investigate and fix. Make sure you have the most recent version of tests (I couldn't update as the conflict blocks merging, and the conflict looks like something relevant to Ria).

- @Niklewa Please add log10-and-then-sigmoid method. Add to experiments, see what happens for instance with standarization_comparison_tool('ethnic_composition', 'white'), as in the notebook.

- The function introduced in the cleaning utils should be the general one defined by Nikodem, which already includes the one defined by Emily as an option (@Niklewa  can you double check this and make it default?) One reason is that we will *urgently* need to make log10-sigmoid transformations as well for some of the variables. 

- Currently there are no tests for the transformation. Upon revision, please write method-parametrized tests for the new function.

- add to notebook experiments for each one of different variable types to the notebook, notably including some funding information, some percentage variables and some outcome variables. We need to shape the pipeline by deciding which variables need a log10 transform before sigmoid, and once we decide by eyeballing the distros, we will hard-code this in the cleaning functions.

**Comment by emackev** [comment]
> * Since the previous target has been merged into `main`, this PR now targets `main`.
> * This PR has a `fips_query.py` conflict, which looks like it's related to @riadas modification to plotting functions. Ria, can you resolve it to your liking?
> * Once the conflict is resolved, please `pull origin main` after making sure your local copy of main is up to date. Then run the test.
> * As the PR stands, it fails tests locally. Investigate and fix. Make sure you have the most recent version of tests (I couldn't update as the conflict blocks merging, and the conflict looks like something relevant to Ria).
> * @Niklewa Please add log10-and-then-sigmoid method. Add to experiments, see what happens for instance with standarization_comparison_tool('ethnic_composition', 'white'), as in the notebook.
> * The function introduced in the cleaning utils should be the general one defined by Nikodem, which already includes the one defined by Emily as an option (@Niklewa  can you double check this and make it default?) One reason is that we will _urgently_ need to make log10-sigmoid transformations as well for some of the variables.
> * Currently there are no tests for the transformation. Upon revision, please write method-parametrized tests for the new function.
> * add to notebook experiments for each one of different variable types to the notebook, notably including some funding information, some percentage variables and some outcome variables. We need to shape the pipeline by deciding which variables need a log10 transform before sigmoid, and once we decide by eyeballing the distros, we will hard-code this in the cleaning functions.

- I merged in main, and I'm taking @riadas 's version of the 2-line plotting change
- All tests still pass locally for me
- The remaining points about adding additional features that @Niklewa was working on may be best handled as a separate PR by Nikodem. The scope of the current PR is fixing the non-standard min-max issue in the standardize-and-scale function. The log10 transforms and other features and notebooks Nikodem was working on could separately be valuable added functionality.

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
If you insist we can do it in two stages. If that's the case, please add a meaningful pytest test for your function to `test_cleaning_utils.py`, as it currently is not tested automatically.

**Comment by emackev** [comment]
> If you insist we can do it in two stages. If that's the case, please add a meaningful pytest test for your function to `test_cleaning_utils.py`, as it currently is not tested automatically.

Thanks. Could you please explain what tests you want beyond what's already there? 
```def test_standardize_and_scale():
    gdp_scaled = standardize_and_scale(gdp)

    for column in gdp_scaled.columns[2:]:
        assert np.min(gdp_scaled[column]) >= -1
        assert np.max(gdp_scaled[column]) <= 1

    assert gdp.shape == gdp_scaled.shape
```
Happy to meet briefly if that'd help.

**Review by Niklewa — APPROVED** [review]
@emackev, I have added testing for NAs in the outcome of `standardize_and_scale`. I do not have more ideas about the tests. Standardized files are also tested by `test_data_grabber`.

As far as I understand,  I should work on the issues listed by @rfl-urbaniak in an another PR? If so, then I have nothing more to add here.
